services/datamanager/builtin/builtin.go

```diff
@@ -569,26 +571,39 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 					if svc.syncSensor != nil && svc.selectiveSyncEnabled {
 						shouldSync = readyToSync(cancelCtx, svc.syncSensor, svc.logger)
 					}
+					svc.lock.Unlock()
+
 					if shouldSync {
-						svc.sync()
+						svc.sync(cancelCtx)
 					}
+				} else {
+					svc.lock.Unlock()
 				}
-				svc.lock.Unlock()
 			}
 		}
 	})
 }
 
-func (svc *builtIn) sync() {
+func (svc *builtIn) sync(cancelCtx context.Context) {
 	svc.flushCollectors()
 
+	svc.lock.Lock()
 	toSync := getAllFilesToSync(svc.captureDir, svc.fileLastModifiedMillis)
 	for _, ap := range svc.additionalSyncPaths {
 		toSync = append(toSync, getAllFilesToSync(ap, svc.fileLastModifiedMillis)...)
 	}
-	for _, p := range toSync {
-		svc.syncer.SyncFile(p)
-	}
+	svc.lock.Unlock()
+
+	goutils.PanicCapturingGo(func() {
```

Comment:
Got this working on a battery test of scenarios: combinations of of reconfiguring different fields while capturing and syncing a large batch of files, small number of files, over a long period of time, etc. 

Wrapped the core sync() logic in a Goroutine since otherwise, this will be blocked while `SyncFile` is waiting to kick off another Goroutine, and thus the ticker on line 564 will be stuck from queueing up any new files. Thought through whether we'd actually want to pause here until the previous tick's goroutines were done, but need to verify that's the resulting behavior.

This doesn't work _super_ well since if I increment `svc.backgroundWorkers` for this Goroutine, it doesn't get decremented while `SyncFile` is stuck, but it does work behavior-wise :)